src/test/java/edition/academy/seventh/persistence/BookServiceTestIT.java

```diff
@@ -0,0 +1,64 @@
+package edition.academy.seventh.persistence;
+
+import edition.academy.seventh.persistence.response.BookDto;
+import edition.academy.seventh.persistence.response.BookstoreBookDto;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.TestPropertySource;
+
+import java.util.List;
+
+import static junit.framework.TestCase.assertNull;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@SpringBootTest
+@TestPropertySource(locations = "classpath:test.properties")
+class BookServiceTestIT {
+
+  @Autowired private BookService bookService;
+
+  @ParameterizedTest(name = "{index} => bookDtos: {0}")
+  @MethodSource(
+      "edition.academy.seventh.persistence.DataProviderForBookServiceTestIT#dataForAddingBookDtos")
+  void should_returnOneMoreBook_when_addOneBook(List<BookDto> bookDtos) {
+    // Given
+    List<BookDto> booksBeforeAdding = bookService.getBooksFromDatabase();
+
+    // When
+    bookService.addBooksToDatabase(bookDtos);
+    List<BookDto> booksAfterAdding = bookService.getBooksFromDatabase();
+
+    // Then
+    assertEquals(booksBeforeAdding.size() + 1, booksAfterAdding.size());
```

Comment:
Co to za magiczne +1?

Reply:
Może użyć assetNotEqual albo jakoś wcześniej określić, że ten numer to +1 książka w bazie. Może jest jeszcze jakaś inna sensowna assercja dla tego przypadku?